### ISSUE TITLE
Locks up (usually) when there are more namespaces than threads

### ISSUE THREAD

**Issue by eraserhd** [weavejester/eftest#71]
This happens when "multithread namespaces" is on.

In VisualVM, I see that all executer threads are parked, waiting on `.get` of `FutureTask`.

Given the stack traces, I'm confident that the problem is that every executer thread is trying to run a namespace in parallel, submitting back to the executer, but there are no threads left in the executer to run tests.

Note that this means that a fair number of threads are idle while running the tests.

Work-around: for now, disable multithread namespaces.
A possible solution: Use a separate thread pool with no limit `(Executors/newCachedThreadPool)` for running namespaces.  Or somehow coordinate differently?

**Comment by eraserhd** [comment]
An easier and nicer work-around is to pass `:thread-count` to `eftest.runner/run-tests`, and compute it by adding the number of namespaces in the vars to the number of threads you want to run tests.

This works well for me:

```clojure
(defn- thread-count
  "eftest can lock up if we have more namespaces than threads (see
  https://github.com/weavejester/eftest/issues/71); however, the threads
  which handle namespaces are basically idle.  So we add threads equal to
  the number of namespaces."
  [vars]
  (let [ns-count     (count (group-by (comp :ns meta) vars))]
    (+ ns-count (.availableProcessors (Runtime/getRuntime)))))

(defn run-tests
  "Run tests in the project.  Without arguments, runs all tests.  Arguments
  are symbols identifying either namespaces or individual tests."
  ([] (run-tests "test"))
  ([& symbols]
   (let [vars         (eftest/find-tests symbols)]
     (eftest/run-tests vars {:test-warn-time 40000
                             :thread-count   (thread-count vars)}))))
```
